app/config.go

```diff
@@ -42,85 +44,149 @@ func readConfigFromConfigFile(configFile string) bool {
+			}
+		}
+		if !foundValidatorAddress {
+			log.Fatalf("[CONFIG] Ethereum.ValidatorAddresses does not contain validator address")
+		}
+
 	}
 
 	// pocket
-	if Config.Pocket.RPCURL == "" {
-		log.Fatal("[CONFIG] Pocket.RPCURL is required")
-	}
-	if Config.Pocket.ChainId == "" {
-		log.Fatal("[CONFIG] Pocket.ChainId is required")
-	}
-	if Config.Pocket.RPCTimeoutMillis == 0 {
-		log.Fatal("[CONFIG] Pocket.RPCTimeoutMillis is required")
-	}
-	if Config.Pocket.PrivateKey == "" {
-		log.Fatal("[CONFIG] Pocket.PrivateKey is required")
-	}
-	if Config.Pocket.TxFee == 0 {
-		log.Fatal("[CONFIG] Pocket.TxFee is required")
-	}
-	if Config.Pocket.VaultAddress == "" {
-		log.Fatal("[CONFIG] Pocket.VaultAddress is required")
-	}
-	if Config.Pocket.MultisigPublicKeys == nil || len(Config.Pocket.MultisigPublicKeys) == 0 {
-		log.Fatal("[CONFIG] Pocket.MultisigPublicKeys is required")
-	}
 
-	// services
-	if Config.MintMonitor.Enabled && Config.MintMonitor.IntervalMillis == 0 {
-		log.Fatal("[CONFIG] MintMonitor.Interval is required")
-	}
-	if Config.MintSigner.Enabled && Config.MintSigner.IntervalMillis == 0 {
-		log.Fatal("[CONFIG] MintSigner.Interval is required")
-	}
-	if Config.MintExecutor.Enabled && Config.MintExecutor.IntervalMillis == 0 {
-		log.Fatal("[CONFIG] MintExecutor.Interval is required")
-	}
-	if Config.BurnMonitor.Enabled && Config.BurnMonitor.IntervalMillis == 0 {
-		log.Fatal("[CONFIG] BurnMonitor.Interval is required")
-	}
-	if Config.BurnSigner.Enabled && Config.BurnSigner.IntervalMillis == 0 {
-		log.Fatal("[CONFIG] BurnSigner.Interval is required")
+	{
+		// cosmos
+		if Config.Pocket.StartHeight == 0 {
+			log.Warn("Pocket.StartBlockHeight is 0")
+		}
+		if Config.Pocket.Confirmations == 0 {
+			log.Warn("Pocket.Confirmations is 0")
+		}
+		if Config.Pocket.GRPCEnabled {
+			if Config.Pocket.GRPCHost == "" {
+				log.Fatal("Pocket.GRPCHost is required when GRPCEnabled is true")
+			}
+			if Config.Pocket.GRPCPort == 0 {
+				log.Fatal("Pocket.GRPCPort is required when GRPCEnabled is true")
+			}
+		} else {
+			if Config.Pocket.RPCURL == "" {
+				log.Fatal("Pocket.RPCURL is required when GRPCEnabled is false")
+			}
+		}
+		if Config.Pocket.RPCTimeoutMillis == 0 {
+			log.Fatal("Pocket.TimeoutMS is required")
+		}
+		if Config.Pocket.ChainID == "" {
+			log.Fatal("Pocket.ChainID is required")
+		}
+		if Config.Pocket.TxFee == 0 {
+			log.Warn("Pocket.TxFee is 0")
+		}
+		if Config.Pocket.Bech32Prefix == "" {
+			log.Fatal("Pocket.Bech32Prefix is required")
+		}
+		if Config.Pocket.CoinDenom == "" {
+			log.Fatal("Pocket.CoinDenom is required")
+		}
+		if !common.IsValidBech32Address(Config.Pocket.Bech32Prefix, Config.Pocket.MultisigAddress) {
+			log.Fatal("Pocket.MultisigAddress is invalid")
+		}
+		if len(Config.Pocket.MultisigPublicKeys) <= 1 {
+			log.Fatal("Pocket.MultisigPublicKeys is required and must have at least 2 public keys")
+		}
+
+		if Config.Pocket.Mnemonic == "" && Config.Pocket.GcpKmsKeyName == "" {
```

Comment:
Actually, the mnemonic is not particularly useful in this implementation. It only adds unnecessary overhead to maintain the source code. I couldn’t understand why we should derive both Ethereum and Cosmos (Pocket) addresses from a mnemonic when we are using the Cosmos private key for Cosmos and `Config.Ethereum.PrivateKey` for the Ethereum side.